src/inmanta/module.py

```diff
@@ -2252,11 +2256,14 @@ def ensure_versioned(self) -> None:
         if not os.path.exists(os.path.join(self.path, ".git")):
             LOGGER.warning("Module %s is not version controlled, we recommend you do this as soon as possible.", self.name)
 
-    @lru_cache()
     def get_ast(self, name: str) -> Tuple[List[Statement], BasicBlock]:
         if self._project is None:
             raise ValueError("Can only get module's AST in the context of a project.")
 
+        # Check local cache
+        if hit := self._ast_cache.get(name, None) is not None:
```

Comment:
1. I wouldn't use the walrus operator here
2. this is wrong, I think, as it caches the result of the comparison, this would make all compiler tests go up in smoke. 

(I think this clearly make the point against using it?)

Reply:
This is only the un-caching part, the caching part happens L2283

Reply:
I think I misinterpreted your comment, but all the compiler tests are indeed going up into smoke. I will change it !